addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3982,26 +3967,21 @@ var SnippetsMenu = Widget.extend({
         const $snippet = $(ev.target).closest('.oe_snippet');
         const snippetId = parseInt(ev.currentTarget.dataset.snippetId);
         ev.stopPropagation();
-        new Dialog(this, {
-            size: 'medium',
-            title: _t('Confirmation'),
-            $content: $('<div><p>' + _t("Are you sure you want to delete the snippet: %s?", $snippet.attr('name')) + '</p></div>'),
-            buttons: [{
-                text: _t("Yes"),
-                close: true,
-                classes: 'btn-primary',
-                click: async () => {
-                    await this.orm.call("ir.ui.view", "delete_snippet", [], {
-                        'view_id': snippetId,
-                        'template_key': this.options.snippets,
-                    });
-                    await this._loadSnippetsTemplates(true);
-                },
-            }, {
-                text: _t("No"),
-                close: true,
-            }],
-        }).open();
+        const message = _t("Are you sure you want to delete the snippet %s?", $snippet[0].getAttribute("name"));
+        this.call("dialog", "add", ConfirmationDialog, {
+            body: markup(`<div><p>${message}</p></div>`),
+            confirm: async () => {
+                await this.orm.call("ir.ui.view", "delete_snippet", [], {
+                    'view_id': snippetId,
+                    'template_key': this.options.snippets,
+                });
+                await this._loadSnippetsTemplates(true);
+                return true;
```

Comment:
you don t need to return something (You juste need to return false if the disalog shouldn't close)